src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -397,5 +397,76 @@ private static unsafe void Populate83FileNameFromRandomBytes(byte* bytes, int by
             chars[10] = s_base32Char[(bytes[6] & 0x1F)];
             chars[11] = s_base32Char[(bytes[7] & 0x1F)];
         }
+
+        /// <summary>
+        /// Create a relative path from one path to another. Paths will be resolved before calculating the difference.
+        /// Default path comparison for the active platform will be used (OrdinalIgnoreCase for Windows, Ordinal for Unix/Mac).
+        /// </summary>
+        /// <param name="relativeTo">The source path the output should be relative to. This path is always considered to be a directory.</param>
+        /// <param name="path">The destination path.</param>
+        /// <returns>The relative path or <paramref name="path"/> if the paths don't share the same root.</returns>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="relativeTo"/> or <paramref name="path"/> is null or an empty string.</exception>
+        public static string GetRelativePath(string relativeTo, string path)
+        {
+            return GetRelativePath(relativeTo, path, PathInternal.DefaultPathComparison);
+        }
+
+        private static string GetRelativePath(string relativeTo, string path, StringComparison comparisonType)
+        {
+            if (string.IsNullOrEmpty(relativeTo)) throw new ArgumentNullException(nameof(relativeTo));
+            if (string.IsNullOrWhiteSpace(path)) throw new ArgumentNullException(nameof(path));
+            if (comparisonType != StringComparison.Ordinal && comparisonType != StringComparison.OrdinalIgnoreCase)
```

Comment:
Since it looks like this is currently a private method, perhaps `Debug.Assert` would be more appropriate.